app/controllers/auth_services_controller.rb

```diff
@@ -70,4 +73,25 @@ def omniauth_providers
   def redirect_path
     :services
   end
+
+  def is_viable_redirect_location(url)
+    # When you visit an attendable page (e.g. a Workshop, Eventor  Meeting) you
+    # probably want to sign-up to that thing, which requires signing in. When
+    # you sign in you want to get sent back to the page you were on previously
+    # so you can click the attend button right away rather than being sent to
+    # your dashboard then having to navigate back to the Workshop/Event/Meeting
+    # page.
+    begin
+      route = Planner::Application.routes.recognize_path(url)
+
+      return true if route[:controller] == "workshops" && route[:action] == "show"
+      return true if route[:controller] == "meetings" && route[:action] == "show"
+      return true if route[:controller] == "events" && route[:action] == "show"
```

Comment:
That feels a bit scary as we're mapping to a "controller + action" pairing. It's a happy coincidence atm that all the actions are called "show" but if that were to change later then that'd make the logic for this more complicated.

The duplication is acceptable as it makes this easier to parse imo.